torch/_tensor_docs.py

```diff
@@ -2103,8 +2103,16 @@ def add_docstr_all(method, docstr):
 
 .. math::
 
-    f(X=k) = (1 - p)^{k - 1} p
+    P(X=k) = (1 - p)^{k - 1} p, k = 1, 2, ...
 
+.. note::
+  Geometric distribution can be defined by two different probability mass functions (PMFs), and the meaning of geometric random variable is determined by the PMF used. 
+  
+  The most common PMF is that used for here where :math:`k` stands for the number of trials before the first success (i.e., :math:`k`-th trial is the first success). Hence, its support is positive integer [1, :math:`\inf`).
+  
+  Another PMF is a shifted version of the above PMF and used for :func:`torch.distributions.geometric.Geometric`:
+  :math:`P(X=k) = (1-p)^{k} p, k = 0, 1, ...`
+  where :math:`k` stands for the number of failures before the first success (i.e., :math:`(k+1)`-th trial is the first success). Hence, its support is non-negative integer [0, :math:`\inf`). 
```

Comment:
The  `k = 1, 2, ...` and `k = 0, 1, ...` ranges are great.  Can we compress the note into a single sentence, say

> `torch.tensor.geometric_()` is one-based and draws samples in `{1, 2, ...}`, whereas `torch.distributions.Geometric` is zero-based and draws samples in `{0, 1, ...}`.

Reply:
Thanks @fritzo! How does the following sound? 

> `torch.Tensor.geometric_` `k`-th trial is the first success hence draws samples in `[1 , inf)`, whereas
`torch.distributions.Geometric` `k+1`-th trial is the first success hence draws samples in `[0, inf)`

Reply:
Sure, that sounds good. I'd use `{0, 1, 2, ...}` and `{1, 2, 3, ...}` rather than intervals to make it clear the sets are integers rather than real intervals. Thanks for clarifying!

Reply:
> I'd use {0, 1, 2, ...} and {1, 2, 3, ...} rather than intervals to make it clear the sets are integers rather than real intervals. 

Good point! Thanks, updated.